src/test/kotlin/ecommerce/product/ProductControllerTest.kt

```diff
@@ -0,0 +1,164 @@
+package ecommerce.product
+
+import io.restassured.RestAssured
+import io.restassured.http.ContentType
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.HttpStatus
+import org.springframework.test.annotation.DirtiesContext
+
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+class ProductControllerTest {
+    @Test
+    fun `create() should be able return 'created 201' response`() {
+        val response =
+            RestAssured
+                .given().log().all()
+                .body(AMERICANO)
+                .contentType(ContentType.JSON)
+                .`when`().post("/api/products")
+                .then().log().all().extract()
+
+        assertThat(response.statusCode()).isEqualTo(HttpStatus.CREATED.value())
+    }
+
+    @Test
+    fun `read() should be able to read a product and return 'ok 200' response`() {
+        createProduct(AMERICANO)
+
+        val response =
+            RestAssured
+                .given().log().all()
+                .contentType(ContentType.JSON)
+                .`when`().get("/api/products")
+                .then().log().all().extract()
+
+        assertThat(response.statusCode()).isEqualTo(HttpStatus.OK.value())
+        assertThat(response.jsonPath().getList("", Product::class.java)).hasSize(1)
```

Comment:
Create precise tests by comparing each response value one by one!

Reply:
I think it’s fine to just list them out without extracting them into separate methods 😊